packages/services/pkg/grpc/relay.go

```diff
@@ -289,13 +295,31 @@ func (server *ecsRelayServer) HandlePushRequest(request *pb.PushRequest) error {
 		Name: recoveredAddress,
 	}
 
+	// Get the client object for this identity to make sure it's authenticated.
 	client, err := server.GetClientFromIdentity(identity)
 	if err != nil {
 		return err
 	}
 
-	// Verify that the message is OK to relay.
+	// Check if the authenticated client has a balance. We permit pushes of messages only for
+	// clients which have a non-zero balance.
+	balance, err := eth.GetCurrentBalance(server.ethClient, recoveredAddress)
```

Comment:
little nits from me: if we get 10 messages per second per client with sufficient balance we probably need to cache this and only refresh it every so often (if no balance, maybe refresh every 5 second; and if balance refresh it every 60s? so we have a maximum of 5s of messages dropped if you just received a drip on a new account; and a minute of "free" messages if you use to have fund)

Reply:
agree, otherwise we spam RPC again

Reply:
addressed in commit 1d476d5440a722e930eb02c47ddd475aa6487161